File: frontend/src/types/product.ts
export interface User {
  id: number;
  nickname?: string | null;
  email?: string | null;
  points?: number;
  coupon_count?: number;
  membership_grade?: string;
  wish_count?: number;
  sales_count?: number;
}

export interface Product {
  id: number;
  seller_id: number;
  seller?: User | null;
  title: string;
  subtitle?: string | null;
  description?: string | null;
  price: number;
  retail_price?: number | null;
  discount_rate?: number | null;
  status: 'AVAILABLE' | 'RESERVED' | 'SOLD';
  images?: string;
  brand?: string | null;
  brand_id?: number | null;
  category_main?: string | null;
  category_medium?: string | null;
  category_small?: string | null;
  condition?: string | null;
  tags?: string | null;
  badge?: string | null;
  is_ready?: boolean;
  has_certificate?: boolean;
  year?: string | null;
  views: number;
  likes: number;
  chat_count: number;
  created_at: string;
  updated_at: string;
}

export interface Category {
  id: number;
  name: string;
  parent_id: number | null;
  emoji?: string | null;
  children: Category[];
}

export interface Brand {
  id: number;
  name: string;
  latin?: string;
  logo?: string;
  popular?: string[];
}

export interface Banner {
  id: number;
  badge: string;
  title: string;
  sub: string;
  bg: string;
  image: string;
  accent: string;
}

export interface QuickAction {
  id: number;
  icon: string;
  title: string;
  subtitle?: string | null;
  bg: string;
  fg?: string;
}

export interface HomePayload {
  banners: Banner[];
  quick_actions: QuickAction[];
  marquee: string[];
  new_arrivals: Product[];
  categories: Category[];
}

export interface ExploreSection {
  brand: Brand;
  products: Product[];
}

export interface ExplorePayload {
  categories: Category[];
  brands: Brand[];
  sections: ExploreSection[];
}

export interface WikiEntry {
  id: number;
  product_name: string;
  product_sub: string;
  price: number;
  traded_at: string;
  image: string;
}

export interface TradeReview {
  id: number;
  title: string;
  author: string;
  rating: number;
  summary: string;
  created_at: string;
}

export interface WantedPost {
  id: number;
  brand: string;
  product: string;
  budget: number;
  note: string;
  created_at: string;
}

export interface Notice {
  id: number;
  tag: string;
  title: string;
  body: string;
  pinned: boolean;
  created_at: string;
}

export interface LoungePayload {
  wiki: WikiEntry[];
  reviews: TradeReview[];
  wanted: WantedPost[];
  notices: Notice[];
}

export interface Letter {
  id: number;
  issue: string;
  title: string;
  excerpt: string;
  cover: string;
  created_at: string;
  is_new?: boolean;
}

File: backend/app/models/product.py
from sqlalchemy import Column, Integer, String, Text, ForeignKey, DateTime, Enum
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.core.database import Base
import enum

class ProductStatus(str, enum.Enum):
    AVAILABLE = "AVAILABLE"
    RESERVED = "RESERVED"
    SOLD = "SOLD"

class Product(Base):
    __tablename__ = "products"

    id = Column(Integer, primary_key=True, index=True)
    seller_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    title = Column(String(255), nullable=False, index=True)
    description = Column(Text, nullable=True)
    price = Column(Integer, nullable=False) # KRW usually without decimals
    status = Column(String(50), default=ProductStatus.AVAILABLE, nullable=False)
    images = Column(Text, nullable=True) # Stored as JSON string
    
    # New Fields
    category_main = Column(String(100), nullable=True)
    category_medium = Column(String(100), nullable=True)
    category_small = Column(String(100), nullable=True)
    condition = Column(String(50), nullable=True)
    tags = Column(Text, nullable=True) # JSON or comma-separated
    
    views = Column(Integer, default=0)
    likes = Column(Integer, default=0)
    chat_count = Column(Integer, default=0)
    
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

    seller = relationship("User", backref="products")
